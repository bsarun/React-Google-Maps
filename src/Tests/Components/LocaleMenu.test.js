import React from 'react';
import LocaleMenu from '../../Components/LocaleMenu';
import SegmentedControl from '../../Components/Inputs/SegmentedControl';
import { IntlProvider } from 'react-intl';
import i18n from '../../i18n.en.json';
import renderer from 'react-test-renderer';
import {shallow, mount, render} from 'enzyme';
import App from '../../Containers/App.jsx';


describe('Todo component renders the todo correctly', () => {
	it('Renders correctly', () => {
		var output = renderer.create(<IntlProvider locale="en" messages={i18n}><LocaleMenu name="test"/></IntlProvider>);
		let tree = output.toJSON();
		expect(tree).toMatchSnapshot();	
	}); 
	it('calls "handleLanguageChange()" on button click - using prototype', () => {
		const mockedEvent = { target: {} ,preventDefault:()=>{}}; 
		const wrapper = mount(<IntlProvider locale="en" messages={i18n}><LocaleMenu name="test"/></IntlProvider>);
		wrapper.find('#locale-options-0').simulate('click', mockedEvent );
		
	});
	it('calls "handleLocaleMenuClick()" on button click - using prototype', () => {
		const mockedEvent = { target: {} ,preventDefault:()=>{}}; 
		const wrapper = mount(<IntlProvider locale="en" messages={i18n}><LocaleMenu name="test"/></IntlProvider>);
		wrapper.find('.locale-menu > button').simulate('click', mockedEvent );
	
	});
});
