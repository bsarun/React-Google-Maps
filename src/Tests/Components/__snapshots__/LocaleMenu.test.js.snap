// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component renders the todo correctly Renders correctly 1`] = `
<div
  className="locale-menu en"
>
  <button
    className="btn btn-toggle hidden-sm hidden-md hidden-lg"
    onClick={[Function]}
  >
    <span
      className="sr-only"
    >
      <span>
        Open language menu
      </span>
    </span>
  </button>
  <form
    className="locale-options"
    onSubmit={[Function]}
  >
    <div
      className="form-group segmented-control"
    >
      
      
      <div
        className="controls"
      >
        <span>
          <button
            className="segmented-option btn selected btn-primary"
            data-value="en"
            disabled={false}
            id="locale-options-0"
            onBlur={[Function]}
            onClick={[Function]}
          >
            <span
              className="wrapper-text"
            >
              English
            </span>
          </button>
        </span>
        <span>
          <button
            className="segmented-option btn btn-default"
            data-value="es"
            disabled={false}
            id="locale-options-1"
            onBlur={[Function]}
            onClick={[Function]}
          >
            <span
              className="wrapper-text"
            >
              Espaï¿½ol
            </span>
          </button>
        </span>
      </div>
      
      
    </div>
  </form>
</div>
`;
